internal/test/e2e/plugins.go

```diff
@@ -254,6 +275,27 @@ var VersionCompatibilityOSSPluginConfigurationTests = []VersionCompatibilityPlug
 		ConfigureForService: true,
 		ConfigureForRoute:   true,
 	},
+	{
+		Name: "pre-function",
+		Config: `{
+			"access": [
+				"kong.log.err('Hello Koko!')"
+			],
+			"functions": []
+		}`,
+		FieldUpdateChecks: map[string][]FieldUpdateCheck{
+			">= 3.0.0": {
+				{
+					Field: "access",
+					Value: []string{
+						"kong.log.err('Hello Koko!')",
+					},
+				},
+			},
+		},
+		ConfigureForService: true,
+		ConfigureForRoute:   true,
+	},
```

Comment:
These two new additions to plugins.go for `post-function` and `pre-function` won't work as the plugins will already be configured. This end-to-end test is required, but it will need to be in it's own test; e.g. `TestVersionCompatibility_EnsureAccessFieldAreNotOverriddenWithEmptyFields` and only if the data plane >= 3.0.